autode/species/species.py

```diff
@@ -1198,5 +1192,60 @@ def explicitly_solvate(self,
         self.solvent.randomise_around(self)
         return None
 
+    def calc_hessian(self,
+                     method:                  ElectronicStructureMethod,
+                     keywords:                Union[Sequence[str], str, None] = None,
+                     numerical:               bool = False,
+                     use_central_differences: bool = False,
+                     num_delta:               Union[float, val.Distance] = val.Distance(2E-3, units='Å')
+                     ) -> None:
+        """
+        Calculate the Hessian
+
+        -----------------------------------------------------------------------
+        Arguments:
+            method: Method to use to calculate the Hessian
+
+            keywords: Keywords to use to calculate the Hessian, or gradient if
+                      numerical = True
+
+            numerical: Whether to do a numerical frequency calculation using
+                       analytic gradients
+
+            use_central_differences: Use central differences to calculate the
+                                 numerical Hessian. If True then use
+                                df/dx = [f(x+h) - f(x-h)] / 2h
+                                otherwise use single sided differences (faster
+                                but less accurate)
+                                df/dx = [f(x+h) - f(x)] / h
+
+            num_delta: Shift applied to each Cartesian coordinate (h) in the
+                       calculation of the numerical Hessian
+        """
+
+        if numerical:
```

Comment:
is there a reason this isn't two different functions, one for numerical and one for not?

Reply:
I don't like one function effectively doing two separate things depending on that bool, it just feels like it should be two functions.
[This ](https://understandlegacycode.com/blog/what-is-wrong-with-boolean-parameters/) explains what I mean a bit better

Reply:
Hmm, it's still doing the same thing but in two different ways (I think most of the QM packages just have a flag in the input files). Using a named parameter elsewhere in the code I think makes it as readable as two functions. I guess the only disadvantage is that the default behaviour isn't obvious unless you look at the signature, but I'm willing to accept that for simplicity (for what's going to be an entirely user-facing method)